api.md

```diff
@@ -468,6 +582,60 @@ CREATE INDEX ON conditions(time, location) USING brin
   WITH (timescaledb.transaction_per_chunk);
 ```
 
+## delete_data_node() :community_function: [](delete_data_node)
+
+This function will remove the data node locally. This will *not*
+affect the remote database in any way, it will just update the local
+index over all existing data nodes.
+
+The data node will be detached from all hypertables that are using
+it. For more information, see [`detach_data_node`].
+
+#### Errors
+
+* An error will be generated if the data node cannot be detached from
+  some hypertable.
+
+#### Required Arguments [](delete_data_node-required-arguments)
+
+| Name        | Description            |
+| ----------- | -----------            |
+| `node_name` | Name of the data node. |
+
+##### Parameter Description
+
+* `node_name` - Name to use for the node locally.
+
+#### Optional Arguments [](delete_data_node-optional-arguments)
+
+| Name        | Description                                           |
+|-------------|-------------------------------------------------------|
+| `if_exists` | Don't generate error if the data node does not exist. |
+| `cascade`   | Cascade the delete.                                   |
+| `force`     | Force removal of data nodes from hypertables.         |
+
+##### Parameter Description
+
+* `if_exists` - Do not generate an error if the data node does not
+  exist. Defaults to false.
+
+* `cascade` - Cascade the delete so that all objects dependent on the
+  data node are deleted as well. Defaults to false.
+
+* `force` - Force removal of data nodes from hypertables, potentially
```

Comment:
Data loss should not be possible here. AFAIK, this is only to force removal even if chunks in some hypertables will be under-replicated (this is the same as for detach, since delete means implicit detaches on all hypertables). We never allow removal in case of data loss as that would leave hypertables in an undefined state (basically broken).

Reply:
Checked the code. You're right, it should only force the operation when there is more than 1 replicated node. Fixed the text.